bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetDevice.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.echonetlite.internal;
+
+import static org.openhab.binding.echonetlite.internal.HexUtil.hex;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Michael Barker - Initial contribution
+ */
+@NonNullByDefault
+public class EchonetDevice extends EchonetObject {
+
+    private final LinkedHashMap<Epc, State> pendingSets = new LinkedHashMap<>();
+    private final HashMap<Epc, State> stateFields = new HashMap<>();
+    private final HashMap<String, Epc> epcByChannelId = new HashMap<>();
+    private final Logger logger = LoggerFactory.getLogger(EchonetDevice.class);
+    @Nullable
+    private EchonetPropertyMap getPropertyMap;
+    private EchonetDeviceListener listener;
+    private boolean initialised = false;
+
+    private long lastPollMs = 0;
+
+    public EchonetDevice(final InstanceKey instanceKey, EchonetDeviceListener listener) {
+        super(instanceKey, Epc.Device.GET_PROPERTY_MAP);
+        this.listener = listener;
+    }
+
+    public void applyProperty(InstanceKey sourceInstanceKey, Esv esv, final int epcCode, final int pdc,
+            final ByteBuffer edt) {
+        final Epc epc = Epc.lookup(instanceKey().klass.groupCode(), instanceKey().klass.classCode(), epcCode);
+
+        if ((Esv.Get_Res == esv || Esv.Get_SNA == esv || Esv.INF == esv) && 0 < pdc) {
+            pendingGets.remove(epc);
+
+            int edtPosition = edt.position();
+
+            final StateDecode decoder = epc.decoder();
+            State state = null;
+            if (null != decoder) {
+                state = decoder.decodeState(edt);
+                if (null == stateFields.put(epc, state)) {
+                    epcByChannelId.put(epc.channelId(), epc);
+                }
+
+                final State pendingState = pendingSets.get(epc);
+                if (null != pendingState && pendingState.equals(state)) {
+                    logger.debug("pendingSet - removing: {} {}", epc, state);
+                    pendingSets.remove(epc);
+                } else if (null != pendingState) {
+                    logger.debug("pendingSet - state mismatch: {} {} {}", epc, pendingState, state);
+                }
```

Comment:
To get rid of the warnings:

`EchonetDevice.java:[69,29] Redundant null check: The variable pendingState cannot be null at this location
EchonetDevice.java:[72,36] Redundant null check: The variable pendingState cannot be null at this location`

```suggestion
                if (pendingSets.containsKey(epc)) {
                    final State pendingState = pendingSets.get(epc);
                    if (pendingState.equals(state)) {
                        logger.debug("pendingSet - removing: {} {}", epc, state);
                        pendingSets.remove(epc);
                    } else {
                        logger.debug("pendingSet - state mismatch: {} {} {}", epc, pendingState, state);
                    }
                }
```

I think the problem is that `LinkedHashMap.get()` will return `null` in two cases: When key is not found and when map contains `null` for given key.

Reply:
This passes spotbugs, but unfortunately looks a bit weird as we've declared a variable `@Nullable` the immediately proceed with using it without a null check.  It also leads to a warning in Intellij.  Using a wrapper method similar to the queue case seems to be the cleaner approach.